server/libs/domain/src/job/job.service.ts

```diff
@@ -89,4 +89,46 @@ export class JobService {
         throw new BadRequestException(`Invalid job name: ${name}`);
     }
   }
+
+  async handleNightlyJobs() {
+    await this.jobRepository.queue({ name: JobName.USER_DELETE_CHECK });
+    await this.jobRepository.queue({ name: JobName.PERSON_CLEANUP });
+    await this.jobRepository.queue({ name: JobName.QUEUE_GENERATE_THUMBNAILS, data: { force: false } });
+  }
+
+  async onDone(item: JobItem) {
```

Comment:
Why there are two switches statement in function?

Reply:
These are the actions to do at the end of each job. Many jobs queue indexing the asset to typesense. Instead of repeating the job.queue line over and over i basically made a big if statement specifically for that job (the first switch). 

The second statement is for all the other jobs that need to get queued.

Reply:
Ah got it

Reply:
Maybe add that as a comment to the code? Would also better explain the missing `break`, because it can be seen as a mistake

Reply:
I added a comment to the `onDone` method and the switch statement. Let me know if that helps with the readability of the code/logic.